alerts/config.go

```diff
@@ -214,31 +641,57 @@ type Threshold ValueWithUnits
 
 // Validate implements structure.Validatable
 func (t Threshold) Validate(v structure.Validator) {
-	v.String("units", &t.Units).OneOf(glucose.MgdL, glucose.MmolL)
-	// This is a sanity check. Client software will likely further constrain these values. The
-	// broadness of these values allows clients to change their own min and max values
-	// independently, and it sidesteps rounding and conversion conflicts between the backend and
-	// clients.
+	v.String("units", &t.Units).OneOf(nontypesglucose.MgdL, nontypesglucose.MmolL)
+	// This is a sanity check. Client software will likely further constrain these
+	// values. The broadness of these values allows clients to change their own min and max
+	// values independently, and it sidesteps rounding and conversion conflicts between the
+	// backend and clients.
 	var max, min float64
 	switch t.Units {
-	case glucose.MgdL, glucose.Mgdl:
-		max = glucose.MgdLMaximum
-		min = glucose.MgdLMinimum
+	case nontypesglucose.MgdL, nontypesglucose.Mgdl:
+		max = nontypesglucose.MgdLMaximum
+		min = nontypesglucose.MgdLMinimum
 		v.Float64("value", &t.Value).InRange(min, max)
-	case glucose.MmolL, glucose.Mmoll:
-		max = glucose.MmolLMaximum
-		min = glucose.MmolLMinimum
+	case nontypesglucose.MmolL, nontypesglucose.Mmoll:
+		max = nontypesglucose.MmolLMaximum
+		min = nontypesglucose.MmolLMinimum
 		v.Float64("value", &t.Value).InRange(min, max)
 	default:
 		v.WithReference("value").ReportError(validator.ErrorValueNotValid())
 	}
 }
 
-// Repository abstracts persistent storage for Config data.
+// Repository abstracts persistent storage in the alerts collection for Config data.
 type Repository interface {
 	Get(ctx context.Context, conf *Config) (*Config, error)
 	Upsert(ctx context.Context, conf *Config) error
 	Delete(ctx context.Context, conf *Config) error
+	List(ctx context.Context, userID string) ([]*Config, error)
 
 	EnsureIndexes() error
 }
+
+// Notification gathers information necessary for sending an alert notification.
+type Notification struct {
+	// Message communicates the alert to the recipient.
+	Message         string
+	RecipientUserID string
+	FollowedUserID  string
+	Sent            func(time.Time)
+}
+
+// RecordsRepository encapsulates queries of the records collection for use with alerts.
+type RecordsRepository interface {
```

Comment:
I think it's important to capture what type of "communications" this repository stores and that those are used solely for alerting purposes. This applies to the struct, interface and collection. Unfortunately, I can't come up with a good suggestion myself.

Reply:
I've taken a stab at a better name. Let me know what you think.

Reply:
Looks good